parser.y

```diff
@@ -286,13 +278,83 @@ unary_expr: postfix_expr { $$ = $1; }
-  /* 6.5.2.4 Postfix increment and decrement operators */
-  | postfix_expr INCR { $$ = std::make_unique<PostfixArithExprNode>(Loc(@1), PostfixOperator::kIncr, $1); }
-  | postfix_expr DECR { $$ = std::make_unique<PostfixArithExprNode>(Loc(@1), PostfixOperator::kDecr, $1); }
+/* 6.5.4 Cast operators*/
+/* TODO: cast type */
+cast_expr: unary_expr { $$ = $1; }
+  ;
+
+/* 6.5.5 Multiplicative operators */
+mul_expr: cast_expr { $$ = $1; }
+  | mul_expr STAR cast_expr { $$ = std::make_unique<BinaryExprNode>(Loc(@2), BinaryOperator::kMul, $1, $3); }
+  | mul_expr DIV cast_expr { $$ = std::make_unique<BinaryExprNode>(Loc(@2), BinaryOperator::kDiv, $1, $3); }
+  | mul_expr MOD cast_expr { $$ = std::make_unique<BinaryExprNode>(Loc(@2), BinaryOperator::kMod, $1, $3); }
+  ;
+
+/* 6.5.6 Additive operators */
+add_expr: mul_expr { $$ = $1; }
+  | add_expr PLUS mul_expr { $$ = std::make_unique<BinaryExprNode>(Loc(@2), BinaryOperator::kAdd, $1, $3); }
+  | add_expr MINUS mul_expr { $$ = std::make_unique<BinaryExprNode>(Loc(@2), BinaryOperator::kSub, $1, $3); }
+  ;
+
+/* 6.5.7 Bitwise shift operators */
+shift_expr: add_expr { $$ = $1; }
+  | shift_expr SHIFT_LEFT add_expr
+  | shift_expr SHIFT_RIGHT add_expr
+  ;
+
+/* 6.5.8 Relational operators */
+relational_expr: shift_expr { $$ = $1; }
+  | relational_expr GT shift_expr { $$ = std::make_unique<BinaryExprNode>(Loc(@2), BinaryOperator::kGt, $1, $3); }
+  | relational_expr LT shift_expr { $$ = std::make_unique<BinaryExprNode>(Loc(@2), BinaryOperator::kLt, $1, $3); }
+  | relational_expr GE shift_expr { $$ = std::make_unique<BinaryExprNode>(Loc(@2), BinaryOperator::kGte, $1, $3); }
+  | relational_expr LE shift_expr { $$ = std::make_unique<BinaryExprNode>(Loc(@2), BinaryOperator::kLte, $1, $3); }
+  ;
+
+/* 6.5.9 Equality operators */
+eq_expr: relational_expr { $$ = $1; }
+  | eq_expr EQ relational_expr { $$ = std::make_unique<BinaryExprNode>(Loc(@2), BinaryOperator::kEq, $1, $3); }
+  | eq_expr NE relational_expr { $$ = std::make_unique<BinaryExprNode>(Loc(@2), BinaryOperator::kNeq, $1, $3); }
+  ;
+
+/* 6.5.10 Bitwise AND operators */
+and_expr: eq_expr
+  | and_expr AMPERSAND eq_expr
+  ;
+
+/* 6.5.11 Bitwise exclusive OR operators */
+exclusive_or_expr: and_expr
+  | exclusive_or_expr XOR and_expr
+  ;
+
+/* 6.5.12 Bitwise inclusive OR operators */
+inclusive_or_expr: exclusive_or_expr
+  | inclusive_or_expr OR exclusive_or_expr
+  ;
+
+/* 6.5.13 Logical AND operators */
+logic_and_expr: inclusive_or_expr
+  | logic_and_expr LOGIC_AND inclusive_or_expr
+  ;
+
+/* 6.5.14 Logical OR operators */
+logic_or_expr: logic_and_expr
+  | logic_or_expr LOGIC_OR logic_and_expr
+  ;
+
+/* 6.5.15 Conditional operators */
+cond_expr: logic_or_expr
+  | logic_or_expr QUESTION expr : cond_expr
+  ;
+
+/* 6.5.16 Assignment operators */
+/* TODO: support multiple assignment operators */
+assign_expr: unary_expr ASSIGN expr {
+    $$ = std::make_unique<SimpleAssignmentExprNode>(Loc(@2), $1, $3);
+  }
+  ;
```

Comment:
I think it's fine for us to complete the grammar at this point:

```suggestion
assign_expr: unary_expr ASSIGN assign_expr {
    $$ = std::make_unique<SimpleAssignmentExprNode>(Loc(@2), $1, $3);
  }
  | cond_expr { $$ = $1; }
  ;
```

Reply:
Thanks! Missed that one 😅